### PR TITLE
Add goose plugin

### PR DESCRIPTION
**Pull request by cblecker**
I blame @ixdy.

cc: @BenTheElder @vllry 

```
$ go test -v ./prow/plugins/goose/... --human --key-path=api-key
=== RUN   TestRealGoose

![goose image](https://images.unsplash.com/photo-1560850262-408d74ecd8d5?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjk0MjU3fQ)
--- PASS: TestRealGoose (2.73s)
=== RUN   TestUrl
--- PASS: TestUrl (0.00s)
=== RUN   TestFormat
--- PASS: TestFormat (0.00s)
=== RUN   TestHttpResponse
--- PASS: TestHttpResponse (0.00s)
=== RUN   TestGeese
--- PASS: TestGeese (0.00s)
PASS
ok      k8s.io/test-infra/prow/plugins/goose    3.609s
```

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
Holding for review

**Comment by vllry** [comment]
This is my "culture impact" at work.

**Comment by justaugustus** [comment]
Extra points for the branch name!
**HONK**
/honk untitled

**Comment by vllry** [comment]
I just realized.

Prow is the goose.

Test failure comments are its honks.

**Comment by cblecker** [comment]
@vllry *steals your mouse*

File: prow/plugins/goose/goose.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (nikhita) [inline prow/plugins/goose/goose.go]: nit: 2019

File: prow/plugins/goose/goose_test.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (nikhita) [inline prow/plugins/goose/goose_test.go]: nit: 2019

**Comment by onlydole** [comment]
Oh wow - this is a masterpiece 😂

**Comment by ixdy** [comment]
For real goose-y behavior, I feel like this should also randomly (with fairly low probability) instead close your PR. :)

**Comment by cblecker** [comment]
@ixdy or run an alpha features test that is guaranteed to honk at you

**Comment by justaugustus** [comment]
@fejta and I were dismayed to find this hadn't merged yet: https://github.com/kubernetes/test-infra/pull/14667#issuecomment-539739560

`/priority critical-urgent`
/honk

File: prow/plugins/goose/goose.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	}
+
+	return fmt.Sprintf("\n![goose image](%s)", img), nil
+}
+
+func (g *realGaggle) URL() string {
+	g.lock.RLock()
+	defer g.lock.RUnlock()
+	uri := string(g.url)
+	if g.key != "" {
```
Comment (fejta) [inline prow/plugins/goose/goose.go]: Do you have an access key we can put in the cluster? It doesn't look like https://api.unsplash.com/photos/random?query=goose works incognito

Comment (fejta) [reply prow/plugins/goose/goose.go]: Okay I have the info I need (and I'm oncall tomorrow), will work on getting this setup.

**Comment by cblecker** [comment]
/hold cancel
:shipit:

**Comment by cblecker** [comment]
📣 https://github.com/kubernetes/test-infra/pull/14691

**Pull request merged**